tests/zfs-tests/tests/functional/online_offline/online_offline_001_pos.ksh

```diff
@@ -60,13 +61,21 @@ function cleanup
 
 	done
 
-	$KILL $killpid >/dev/null 2>&1
+	if $PS --no-headers -p $killpid > /dev/null; then
+		$KILL $killpid >/dev/null 2>&1
+	fi
 	[[ -e $TESTDIR ]] && log_must $RM -rf $TESTDIR/*
 }
 
 log_assert "Turning a disk offline and back online during I/O completes."
 
-$FILE_TRUNC -f $((64 * 1024 * 1024)) -b 8192 -c 0 -r $TESTDIR/$TESTFILE1 &
+if is_linux; then
+	set -x 
+	$DD if=/dev/urandom iflag=fullblock bs=1M count=$((64 * 1024 * 1024)) of="$TESTDIR/$TESTFILE1" &
+	ls -l $TESTDIR/$TESTFILE1
+else
+	$FILE_TRUNC -f $((64 * 1024 * 1024)) -b 8192 -c 0 -r $TESTDIR/$TESTFILE1 &
+fi 
```

Comment:
Wasn't the plan to use `$FILE_WRITE` for this so we wouldn't need any linux specific logic?